### ISSUE TITLE
Exception thrown: 'Microsoft.Bot.Connector.MicrosoftAppCredentials.OAuthException' in mscorlib.dll 

### ISSUE THREAD

**Issue by nmsohn** [ankitbko/SkypeForBusinessBot#5]
My bot builder is the latest version.
The code breaks here:
`await postToBot.PostAsync(activity, CancellationToken.None);`

Is there something that I need to make a change in the activity object?

```
Activity activity = new Activity()
{
From = new ChannelAccount { Id = fromId, Name = fromId },
Conversation = new ConversationAccount { Id = conversationId },
Recipient = new ChannelAccount { Id = "Bot" },
ServiceUrl = "https://skype.botframework.com",
ChannelId = "skype",
ChannelData = SendMessageUrl 
};
```

**Comment by KiranBastawadi** [comment]
Left at the same status. I am wondering if it has to do with BotApp ID registration. Do I have to create my won bot on Azure and use the ID and Bot here to continue ?

**Comment by ankitbko** [comment]
Yes you would need to register the bot and put correct Microsoft app id and secret in the config.

**Comment by rajay1210** [comment]
Hi @ankitbko 

I was using the above mentioned code for linking Bot Framework with Skype for Business using **UCMA (Unified Communications Managed API)**.

Your blog has been of great help. Thanks for that!! :)

I would like to know **where have the credentials been declared in the committed project**.

When I run the code, I am able to use the Skype for Business Channel. But when I replace the Lync Channel with UCMA, I am getting the same issue.

I have tried changing the bot builder and connector versions with no success.

Please guide me on how to solve this issue.
